dupedetection/client.go

```diff
@@ -0,0 +1,127 @@
+package dupedetection
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/pastelnetwork/gonode/common/errors"
+	"github.com/pastelnetwork/gonode/common/log"
+	"github.com/pastelnetwork/gonode/common/random"
+)
+
+type client struct {
+	config *Config
+}
+
+// Generate implements dupedection.Client.Generate
+func (client *client) Generate(ctx context.Context, path string) (*DupeDetection, error) {
+
+	// Copy image file to dupe detection service input directory
+	outputBase, outputPath, err := client.copyImageToInputDir(path)
+	if err != nil {
+		return nil, errors.Errorf("failed to copy image to dupe detection input directory: %w", err)
+	}
+
+	// Collect result from dupe detection output directory
+	result, err := client.collectOutput(ctx, outputBase, outputPath)
+	if err != nil {
+		err = errors.Errorf("failed to collect output from dupe detection service: %w", err)
+	}
+
+	return result, err
+}
+
+func (client *client) copyImageToInputDir(inputPath string) (string, string, error) {
```

Comment:
If we already agree on the rule how "output" file will be named. For example, output = "input.json" then return the expected output file name is more efficient than  walking the directory.